### PR TITLE
Update to NNP-I v1.0.0.5

### PR DESCRIPTION
**Pull request by omromano**
Summary: Align code to NNP-I v1.0.0.5 (glow tracing changes).

### CODE REVIEW

**Comment by omromano** [comment]
Hi @jfix71, @arunm-git,
Please take a look.

File: lib/Backends/NNPI/NNPIMLTraceWrapper.cpp [hunk 1]
```diff
@@ -28,6 +28,16 @@
 #include <unordered_map>
 #include <vector>
 
+/// Dump event entry to stream.
+static void dumpTraceEntry(const NNPITraceEntry &entry,
+                           std::ostream &outstream) {
+  outstream << "HTime:" << entry.hostTime << " ETime:" << entry.engineTime;
+  for (auto param : entry.params) {
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPIMLTraceWrapper.cpp:35]: nit: 
```suggestion
  for (const auto &param : entry.params) {
```

File: lib/Backends/NNPI/NNPIMLTraceWrapper.cpp [hunk 2]
```diff
@@ -58,20 +68,21 @@ static eIceCapsSwTraceEvent swEventTypes[] = {
...
+    : capsSession_(0), devID_(devID), devIDSet_(false), dumpRawEventsPath_() {}
 
 NNPITraceContext::~NNPITraceContext() { destroyInternalContext(); }
 
 bool NNPITraceContext::startCapture(NNPIDeviceContext deviceContext,
                                     bool swTraces, bool hwTraces,
                                     uint32_t softwareBufferSizeMB,
-                                    uint32_t hardwareBufferSizeMB) {
+                                    uint32_t hardwareBufferSizeMB,
+                                    std::string dumpRawEventsPath) {
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPIMLTraceWrapper.cpp:79]: nit:
```suggestion
                                    const std::string &dumpRawEventsPath) {
```

File: lib/Backends/NNPI/NNPIOptions.h [hunk 3]
```diff
@@ -400,6 +400,12 @@ class NNPIDeviceOptions : public NNPIOptions {
       "Set the max internal buffer size for device hardware traces."
       "Enabled only when hardwareTraces=1 (use 0 for hard coded default).",
       "NNPI_HW_TRACES_BUFFER_SIZE", "0");
+  ///
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPIOptions.h:403]: nit: empty comment

File: lib/Backends/NNPI/NNPITracing.cpp [hunk 4]
```diff
@@ -177,30 +189,39 @@ bool NNPIDeviceTracing::addTrace(
...
+                   << " end time " << entry.hostTime;
     }
-    inflight.erase(name);
+    inflight.erase(eventKey);
   } else if (state == "po") {
     traceContext->logTraceEvent(name, TraceLevel::OPERATOR,
                                 TraceEvent::InstantType, entry.hostTime,
                                 entry.params, affinId);
+  } else {
+    if (entry.params.count("engine") > 0 && entry.params["engine"] != "HW") {
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPITracing.cpp:217]: nit:
```suggestion
  } else if (entry.params.count("engine") > 0 && entry.params["engine"] != "HW") {
```

File: lib/Backends/NNPI/NNPITracing.cpp [hunk 5]
```diff
@@ -228,6 +249,19 @@ bool NNPIDeviceTracing::stopAndUpdate(TraceContext *traceContext,
   for (auto entry : traceCtx_->getEntries()) {
     addTrace(entry, inflight, traceContext);
   }
+  if (inflight.size() > 0) {
+    LOG(WARNING) << "[INCOMPLETE EVENT] " << inflight.size()
+                 << " events not logged (still in flight/incomplate)";
+    for (auto event : inflight) {
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPITracing.cpp:255]: nit: 
```suggestion
    for (const auto &event : inflight) {
```

File: lib/Backends/NNPI/NNPITracing.h [hunk 6]
```diff
@@ -63,7 +63,7 @@ class NNPIDeviceTracing {
   /// Start recording events.
   bool start(TraceContext *traceContext, NNPIDeviceContext deviceContext,
              bool swTraces, bool hwTraces, uint32_t softwareBufferSizeMB,
-             uint32_t hardwareBufferSizeMB);
+             uint32_t hardwareBufferSizeMB, std::string dumpRawEventsPath);
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPITracing.h:66]: nit:
```suggestion
             uint32_t hardwareBufferSizeMB, const std::string &dumpRawEventsPath);
```

**Comment by omromano** [comment]
Thanks @jfix71, 
I have made the suggested changes.

**Pull request closed without merging**